float32 position1
float32 position2
float32 position3
float32 position4
